types/hellosign-sdk/index.d.ts

```diff
@@ -420,13 +420,24 @@ declare namespace HelloSign {
     interface TemplateResponse {
         template: Template;
     }
+    interface TemplateFiles {
+        file_type?: 'pdf' | 'zip';
+        get_url?: boolean;
+        get_data_uri?: boolean;
+    }
     interface TemplateModule {
         list(): Promise<{ templates: Template[] }>;
         get(templateId: string): Promise<TemplateResponse>;
         addUser(templateId: string, user: AccountIdOrEmailRequestOptions): Promise<TemplateResponse>;
         removeUser(templateId: string, user: AccountIdOrEmailRequestOptions): Promise<TemplateResponse>;
         createEmbeddedDraft(options: Template): Promise<TemplateResponse>;
         delete(templateId: string): Promise<any>;
+        files(templateId: string, options?: TemplateFiles):
+            Promise<{
```

Comment:
From HelloSign's docs, this method can return 

- PDF or ZIP file or,
- if get_url is set, a JSON object with a url to the file (PDFs only) or,
- If get_data_uri is set, a JSON object with a data_uri representing the base64 encoded file (PDFs only) is returned or,
- if the files are currently being prepared, a status code of 409 will be returned instead.

Reply:
```suggestion
            Promise<Buffer | {
                file_url: string;
                expires_at: Date;
                data_uri: string;
            }>;
```

Reply:
@dvprrsh If I add the suggested change, I then cannot get the return value type. 
This is what VS Code reports
```
Property 'file_url' does not exist on type 'Buffer | { file_url: string; expires_at: Date; data_uri: string; }'.
```

Any suggestion on how this issue can be resolved?

Reply:
I would suggest changing the type to something like this, making the final type dynamic depending on the options passed in.

```suggestion
  type FilesOptions = { file_type: "pdf", get_url?: boolean, get_data_uri?: boolean } | { file_type: "zip" };

  interface TemplateModule {
    ....
    files: <Options extends FilesOptions | undefined>(
      templateId: string,
      options?: Options,
    ) => Promise<
      Options extends { file_type: "zip" } 
        ? Buffer
        : Options extends { file_type: "pdf", get_url: true }
        ? { file_url: string; expires_at: number }
        : Options extends { file_type: "pdf", get_data_uri: true }
        ? { data_uri: string; expires_at: number }
        : Buffer
    >;
```

Reply:
> @dvprrsh If I add the suggested change, I then cannot get the return value type.
> This is what VS Code reports
> 
> ```
> Property 'file_url' does not exist on type 'Buffer | { file_url: string; expires_at: Date; data_uri: string; }'.
> ```
> 
> Any suggestion on how this issue can be resolved?

I would suggest changing the type to something like this, making the return type dynamic depending on the options passed in.

```typescript
  type FilesOptions = { file_type: "pdf", get_url?: boolean, get_data_uri?: boolean } | { file_type: "zip" };

  interface TemplateModule {
    ....
    files: <Options extends FilesOptions | undefined>(
      templateId: string,
      options?: Options,
    ) => Promise<
      Options extends { file_type: "zip" } 
        ? Buffer
        : Options extends { file_type: "pdf", get_url: true }
        ? { file_url: string; expires_at: number }
        : Options extends { file_type: "pdf", get_data_uri: true }
        ? { data_uri: string; expires_at: number }
        : Buffer
    >;
```

Reply:
@dvprrsh Thanks, that did the trick! :)

Reply:
No problem